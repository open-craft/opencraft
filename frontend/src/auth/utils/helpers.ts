import { V2Api } from 'global/api';

export const checkAuthAndRefreshToken = async () => {
  const refreshToken = localStorage.getItem('token_refresh');
  let response = null;

  if (refreshToken) {
    try {
      response = await V2Api.authRefreshCreate({
        data: { refresh: refreshToken }
      });
    } catch (e) {
      return false;
    }
  }

  if (response) {
    localStorage.setItem('token_access', response.access);
    return true;
  }

  return false;
};
