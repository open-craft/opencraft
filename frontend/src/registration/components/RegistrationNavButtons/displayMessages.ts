const messages = {
  back: {
    defaultMessage: 'Back',
    description: 'Text that goes in the back button in the registration form.'
  },
  next: {
    defaultMessage: 'Next',
    description: 'Text that goes in the next button in the registration form.'
  }
};

export default messages;
