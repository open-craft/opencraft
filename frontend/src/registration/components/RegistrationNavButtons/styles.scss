.registration-nav {
  display: inline-block;
  width: 100%;
  padding-top: 32px;

  .registration-nav-btn {
    font-size: 14px;
  }

  .spinner {
    margin-right: 10px;
  }
}
