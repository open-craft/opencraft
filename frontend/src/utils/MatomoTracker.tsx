import React, { ReactNode } from 'react';
import {
  createInstance,
  MatomoProvider,
  useMatomo
} from '@datapunt/matomo-tracker-react';
import {
  MATOMO_ALIAS_DOMAIN,
  MATOMO_BASE_URL,
  MATOMO_MY_DOMAIN,
  MATOMO_SITE_ID
} from '../global/constants';

interface MatomoTrackerProps {
  children: ReactNode;
}

interface MatomoAdditionalConfiguration {
  [propName: string]: any;
}

interface MatomoConfiguration {
  urlBase: string;
  siteId: number;
  configurations?: MatomoAdditionalConfiguration;
}

interface MatomoUserIdTrackerProps {
  userId: string;
}

/*
 * This wrapper component initializes the Matomo tracking by wrapping the children
 * elements, if the tracking is enabled by providing a valid value for the 'MATOMO_BASE_URL'
 * constant. It allows tracking page views, events and the user id, when available.
 *
 * The 'MATOMO_ALIAS_DOMAIN' constant allows the cross-domain linking of another site with
 * this site so that they are treated as the same site in Matomo.
 */
export const MatomoTracker: React.FC<MatomoTrackerProps> = (
  props: MatomoTrackerProps
) => {
  let element = props.children;
  if (MATOMO_BASE_URL) {
    const matomoConfiguration: MatomoConfiguration = {
      urlBase: MATOMO_BASE_URL,
      siteId: parseInt(MATOMO_SITE_ID, 10)
    };
    if (MATOMO_ALIAS_DOMAIN) {
      const myDomain = `*.${MATOMO_MY_DOMAIN}`;
      const aliasDomain = `*.${MATOMO_ALIAS_DOMAIN}`;
      matomoConfiguration.configurations = {
        enableCrossDomainLinking: true,
        setDomains: [myDomain, aliasDomain]
      };
    }
    element = (
      <MatomoProvider value={createInstance(matomoConfiguration)}>
        {props.children}
      </MatomoProvider>
    );
  }
  return <>{element}</>;
};

/**
 * This is a component passes the provided user id to Matomo if Matomo tracking is enabled.
 * The 'pushInstruction' function exposed by the 'useMatomo()' hook is used to do this.
 *
 * This component is needed as all the components in the registration process are classes and hence
 * the 'yseMatomo' hook cannot be directly used in those components.
 */
export const MatomoUserIdTracker: React.FC<MatomoUserIdTrackerProps> = (
  props: MatomoUserIdTrackerProps
) => {
  const { pushInstruction } = useMatomo();
  if (MATOMO_BASE_URL && props.userId) {
    pushInstruction('setUserId', props.userId);
  }
  return null;
};
