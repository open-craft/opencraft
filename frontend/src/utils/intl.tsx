import React from "react";
import { FormattedMessage, MessageDescriptor } from "react-intl";

type MessageMap = { [k: string]: Omit<MessageDescriptor, 'id'> };

interface Props<T, K extends keyof T> {
    id: K;
    messages: T;
    values?: any;
}


export const WrappedMessage = (props: Props<MessageMap, string>) => {
    const {messages, id, ...values} = props;
    return (
        <FormattedMessage  {...messages[id]} id={id} {...values}/>
    );
};
