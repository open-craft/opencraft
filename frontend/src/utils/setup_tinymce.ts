import 'tinymce/tinymce';
import 'tinymce/themes/silver';
import 'tinymce/skins/ui/oxide/skin.min.css';
import 'tinymce/skins/ui/oxide/content.min.css';
import 'tinymce/plugins/advlist';
import 'tinymce/plugins/autolink';
import 'tinymce/plugins/lists';
import 'tinymce/plugins/link';
import 'tinymce/plugins/image';
import 'tinymce/plugins/charmap';
import 'tinymce/plugins/print';
import 'tinymce/plugins/preview';
import 'tinymce/plugins/anchor';
import 'tinymce/plugins/searchreplace';
import 'tinymce/plugins/visualblocks';
import 'tinymce/plugins/code';
import 'tinymce/plugins/fullscreen';
import 'tinymce/plugins/insertdatetime';
import 'tinymce/plugins/media';
import 'tinymce/plugins/table';
import 'tinymce/plugins/paste';
import 'tinymce/plugins/help';
import 'tinymce/plugins/wordcount';
