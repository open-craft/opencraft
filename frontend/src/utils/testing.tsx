import { createMemoryHistory } from 'history';
import * as React from 'react';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router';
import * as renderer from 'react-test-renderer';
import { applyMiddleware, createStore } from 'redux';
import thunk from 'redux-thunk';
import { createRootReducer } from '../global/reducers';

export const setupComponentForTesting = (
  reactContent: JSX.Element,
  storeContents = {}
) => {
  const middleware = applyMiddleware(thunk);

  const store = createStore(
    createRootReducer(createMemoryHistory()),
    storeContents,
    middleware
  );

  return renderer.create(
    <IntlProvider textComponent={React.Fragment} locale="en">
      <Provider store={store}>
        <MemoryRouter>{reactContent}</MemoryRouter>
      </Provider>
    </IntlProvider>
  );
};
