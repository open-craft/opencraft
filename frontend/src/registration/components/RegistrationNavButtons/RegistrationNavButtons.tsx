import * as React from 'react';
import './styles.scss';

import { Button, Spinner } from 'react-bootstrap';
import { WrappedMessage } from 'utils/intl';
import messages from './displayMessages';

interface Props {
  loading: boolean;
  showBackButton: boolean;
  showNextButton: boolean;
  disableNextButton: boolean;
  handleNextClick: Function;
  handleBackClick: Function;
}

export const RegistrationNavButtons: React.SFC<Props> = (props: Props) => {
  return (
    <div className="registration-nav">
      <Button
        className="float-left registration-nav-btn"
        variant="outline-primary"
        size="lg"
        onClick={() => {
          props.handleBackClick();
        }}
      >
        <WrappedMessage messages={messages} id="back" />
      </Button>
      <Button
        className="float-right loading registration-nav-btn"
        variant="primary"
        size="lg"
        disabled={props.disableNextButton || props.loading}
        onClick={() => {
          props.handleNextClick();
        }}
      >
        {props.loading === true && (
          <Spinner animation="border" size="sm" className="spinner" />
        )}
        <WrappedMessage messages={messages} id="next" />
      </Button>
    </div>
  );
};
