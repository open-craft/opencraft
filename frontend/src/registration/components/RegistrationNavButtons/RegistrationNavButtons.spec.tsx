import React from 'react';
import { setupComponentForTesting } from "utils/testing";
import { RegistrationNavButtons } from './RegistrationNavButtons';

it('renders without crashing', () => {
    const tree = setupComponentForTesting(<RegistrationNavButtons />).toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders loading button', () => {
    const tree = setupComponentForTesting(<RegistrationNavButtons loading={true} />).toJSON();
    expect(tree).toMatchSnapshot();
});
