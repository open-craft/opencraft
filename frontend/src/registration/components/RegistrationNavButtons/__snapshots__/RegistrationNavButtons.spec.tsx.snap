// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders loading button 1`] = `
<div
  className="registration-nav"
>
  <button
    className="float-left registration-nav-btn btn btn-outline-primary btn-lg"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Back
  </button>
  <button
    className="float-right loading registration-nav-btn btn btn-primary btn-lg"
    disabled={true}
    onClick={[Function]}
    type="button"
  >
    <div
      className="spinner spinner-border spinner-border-sm"
    />
    Next
  </button>
</div>
`;

exports[`renders without crashing 1`] = `
<div
  className="registration-nav"
>
  <button
    className="float-left registration-nav-btn btn btn-outline-primary btn-lg"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Back
  </button>
  <button
    className="float-right loading registration-nav-btn btn btn-primary btn-lg"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Next
  </button>
</div>
`;
